### ISSUE TITLE
componentWillReceiveProps and componentDidUpdate unexpected behaviour on v2

### ISSUE THREAD

**Issue by 1987cr** [react-navigation/react-navigation#4262]
### Current Behaviour

When switching screens on a TabNavigator, React lifecycle hooks componentWillReceiveProps and componentDidUpdate of the screens are getting triggered when navigating in and out of them.

Navigating from A to B:

- CWRP and CDU from A are triggered.
- CWRP and CDU from B are triggered.

Navigating from B to A:

- CWRP and CDU from A are triggered.
- CWRP and CDU from B are triggered.

This behaviour was not present on v1.

### Expected Behavior

componentWillReceiveProps and componentDidUpdate should be triggered when setting params explicitly to a route.

### How to reproduce

- Have a simple TabNavigator with two react components as screens.
- Put componentWillReceiveProps and componentDidUpdate lifecycle hooks on each component.
- Navigate between them.

See code below:

```
import React, { Component } from "react";
import { StyleSheet, Text, View } from "react-native";
import { createBottomTabNavigator } from "react-navigation";

class ScreenA extends React.Component {
  componentWillReceiveProps(nextProps) {
    console.log("CWRP: Screen A");
  }

  componentDidUpdate(prevProps, prevState) {
    console.log("CDU: Screen A");
  }

  render() {
    return (
      <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
        <Text>Screen A</Text>
      </View>
    );
  }
}

class ScreenB extends React.Component {
  componentWillReceiveProps(nextProps) {
    console.log("CWRP: Screen B");
  }

  componentDidUpdate(prevProps, prevState) {
    console.log("CDU: Screen B");
  }

  render() {
    return (
      <View style={{ flex: 1, alignItems: "center", justifyContent: "center" }}>
        <Text>Screen B</Text>
      </View>
    );
  }
}

const RootStack = createBottomTabNavigator({
  A: ScreenA,
  B: ScreenB
});

export default class App extends Component {
  render() {
    return <RootStack />;
  }
}
```

### Your Environment

| software         | version
| ---------------- | -------
| react-navigation | 2.0.1
| react-native     | 0.53.0

**Comment by craigdub** [comment]
@1987cr Did you find a solution for this?

**Comment by brentvatne** [comment]
this is fixed in latest - 2.2.5

**Comment by brentvatne** [comment]
(you can verify by running the example code provided above in snack - https://snack.expo.io/rJE2OJjl7 -- i added componentDidMount to it also)
